### PR TITLE
test(e2e,playwright): settings page

### PR DESCRIPTION
**Pull request by Riya267**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

 #51705

Snapshot:-
image

### CODE REVIEW

**Comment by Riya267** [comment]
camperbot cant code

File: e2e/settings.spec.ts [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+  let page: Page;
+
+  test.beforeAll(async ({ browser }) => {
+    page = await browser.newPage();
+    await page.goto('/settings');
+  });
+
+  test.afterAll(async () => {
+    await page.close();
+  });
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:66]:
```suggestion
  test.beforeEach(async ({ page }) => {
    await page.goto('/settings');
  });
```

We should ideally run tests in isolation.

File: e2e/settings.spec.ts [hunk 2]
```diff
@@ -0,0 +1,482 @@
+import { test, expect, type Page } from '@playwright/test';
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:1]:
```suggestion
import { test, expect } from '@playwright/test';
```

See below

File: e2e/settings.spec.ts [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+  translations.certification.title['Legacy Back End'],
+  translations.certification.title['Legacy Data Visualization'],
+  translations.certification.title[
+    'Legacy Information Security and Quality Assurance'
+  ]
+];
+
+test.describe('Settings', () => {
+  let page: Page;
+
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:58]:
```suggestion
```

We don't need a global `page` variable because...

File: e2e/settings.spec.ts [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+  test.beforeAll(async ({ browser }) => {
+    page = await browser.newPage();
+    await page.goto('/settings');
+  });
+
+  test.afterAll(async () => {
+    await page.close();
+  });
+
+  test('Should have the correct page title', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:68]:
```suggestion
  test('Should have the correct page title', async ({ page }) => {
```

We can grab `page` directly from the function args.

File: e2e/settings.spec.ts [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+    await expect(page).toHaveTitle(settingsObject.pageTitle);
+  });
+
+  test('Should display the correct header', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:72]:
```suggestion
  test('Should display the correct header', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+    await expect(header).toBeVisible();
+    await expect(header).toContainText(
+      `${translations.settings.for.replace(
+        settingsObject.userNamePlaceholder,
+        settingsObject.certifiedUsername
+      )}`
+    );
+  });
+
+  test('Should validate Username Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:83]:
```suggestion
  test('Should validate Username Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+      .getByTestId(settingsTestIds.usernameSettings)
+      .fill(settingsObject.certifiedUsername);
+    const saveButton = page.getByRole('button', {
+      name: translations.settings.labels.username
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
+  });
+
+  test('Should validate Email Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:141]:
```suggestion
  test('Should validate Email Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+    await expect(noWeeklyBtn).toBeVisible();
+    await noWeeklyBtn.click();
+    await expect(
+      page
+        .getByRole('group', { name: "Send me Quincy's weekly email" })
+        .locator('p')
+    ).toBeVisible();
+  });
+
+  test('Should validate Privacy Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:181]:
```suggestion
  test('Should validate Privacy Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+    await saveButton.press('Enter');
+    await expect(page.getByText(translations.settings.data)).toBeVisible();
+    const downloadButton = page.getByRole('link', {
+      name: translations.buttons['download-data']
+    });
+    await expect(downloadButton).toBeVisible();
+    await downloadButton.click();
+  });
+
+  test('Should validate Internet Presence Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:257]:
```suggestion
  test('Should validate Internet Presence Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+    await expect(
+      page.getByTestId(settingsTestIds.internetPresence)
+    ).toBeVisible();
+    const saveButton = page.getByRole('button', {
+      name: translations.settings.headings.internet
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
+  });
+  test('Should validate Portfolio Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:272]:
```suggestion
  test('Should validate Portfolio Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 11]
```diff
@@ -0,0 +1,482 @@
...
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
+    const removeButton = page.getByRole('button', {
+      name: translations.buttons['remove-portfolio']
+    });
+    await expect(removeButton).toBeVisible();
+    await removeButton.click();
+  });
+
+  test('Should validate Personal Portfolio Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:301]:
```suggestion
  test('Should validate Personal Portfolio Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 12]
```diff
@@ -0,0 +1,482 @@
...
+    );
+    await expect(addPortfolioButton).toBeVisible();
+    await addPortfolioButton.click();
+    const removeButton = page.getByRole('button', {
+      name: translations.buttons['remove-portfolio']
+    });
+    await expect(removeButton).toBeVisible();
+    await removeButton.click();
+  });
+  test('Should validate Accademy Honesty Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:348]:
```suggestion
  test('Should validate Accademy Honesty Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 13]
```diff
@@ -0,0 +1,482 @@
...
+    await expect(
+      page.getByText(
+        translations.settings.honesty.p7.replace(
+          settingsObject.supportEmailPlaceholder,
+          settingsObject.supportEmail
+        )
+      )
+    ).toBeVisible();
+  });
+  test('Should validate Certification Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:389]:
```suggestion
  test('Should validate Certification Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 14]
```diff
@@ -0,0 +1,482 @@
...
+      ).toBeVisible();
+      await expect(
+        page.getByRole('link', {
+          name: `${translations.buttons['show-cert']} ${certifications[i]}`
+        })
+      ).toBeVisible();
+    }
+  });
+
+  test('Should validate Legacy Certification Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:411]:
```suggestion
  test('Should validate Legacy Certification Settings', async ({ page }) => {
```

File: e2e/settings.spec.ts [hunk 15]
```diff
@@ -0,0 +1,482 @@
...
+      ).toBeVisible();
+      await expect(
+        page.getByRole('link', {
+          name: `${translations.buttons['show-cert']} ${legacyCertifications[i]}`,
+          exact: true
+        })
+      ).toBeVisible();
+    }
+  });
+  test('Should validate Danger Section Settings', async () => {
```
Comment (naomi-lgbt) [inline e2e/settings.spec.ts:433]:
```suggestion
  test('Should validate Danger Section Settings', async ({ page }) => {
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: e2e/settings.spec.ts (hunk 1 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 2 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 3 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 4 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 5 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 6 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 7 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 8 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 9 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 10 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 11 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 12 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 13 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 14 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts (hunk 15 above)
Comment (Riya267) [reply e2e/settings.spec.ts]: done

**Review by naomi-lgbt — APPROVED** [review]

File: e2e/settings.spec.ts [hunk 16]
```diff
@@ -0,0 +1,508 @@
...
+    await expect(
+      page.getByTestId(settingsTestIds.internetPresence)
+    ).toBeVisible();
+    const saveButton = page.getByRole('button', {
+      name: translations.settings.headings.internet
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
+  });
+  test('Should validate Portfolio Settings', async ({ page, browserName }) => {
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts [hunk 17]
```diff
@@ -0,0 +1,508 @@
...
+    const addPortfolioButton = page.getByText(
+      translations.buttons['add-portfolio']
+    );
+    await expect(addPortfolioButton).toBeVisible();
+    await addPortfolioButton.click();
+    const removeButton = page.getByRole('button', {
+      name: translations.buttons['remove-portfolio']
+    });
+    await expect(removeButton).toBeVisible();
+    await removeButton.click();
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: removed this click functionality

File: e2e/settings.spec.ts [hunk 18]
```diff
@@ -0,0 +1,508 @@
...
+    const saveButton = page.getByRole('button', {
+      name: translations.buttons['save-portfolio']
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
+    const removeButton = page.getByRole('button', {
+      name: translations.buttons['remove-portfolio']
+    });
+    await expect(removeButton).toBeVisible();
+    await removeButton.click();
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: removed this click functionality

File: e2e/settings.spec.ts [hunk 19]
```diff
@@ -0,0 +1,508 @@
...
+      })
+    ).toBeVisible();
+    await expect(
+      page.getByTestId(settingsTestIds.internetPresence)
+    ).toBeVisible();
+    const saveButton = page.getByRole('button', {
+      name: translations.settings.headings.internet
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: removed this click functionality

File: e2e/settings.spec.ts [hunk 20]
```diff
@@ -0,0 +1,508 @@
...
+      name: translations.settings.headings.privacy
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
+    await expect(page.getByText(translations.settings.data)).toBeVisible();
+    const downloadButton = page.getByRole('link', {
+      name: translations.buttons['download-data']
+    });
+    await expect(downloadButton).toBeVisible();
+    await downloadButton.click();
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: removed this click functionality

File: e2e/settings.spec.ts [hunk 21]
```diff
@@ -0,0 +1,508 @@
...
+      )
+    ).toBeVisible();
+    await page
+      .getByTestId(settingsTestIds.usernameSettings)
+      .fill(settingsObject.certifiedUsername);
+    const saveButton = page.getByRole('button', {
+      name: translations.settings.labels.username
+    });
+    await expect(saveButton).toBeVisible();
+    await saveButton.press('Enter');
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: we checking this action in flash message test so removed it form here

File: e2e/settings.spec.ts [hunk 22]
```diff
@@ -0,0 +1,508 @@
...
+    await expect(
+      page.getByRole('heading', {
+        name: translations.settings.headings.privacy
+      })
+    ).toBeVisible();
+    await expect(page.getByText(translations.settings.privacy)).toBeVisible();
+    await expect(
+      page
+        .locator('p')
+        .filter({ hasText: translations.settings.labels['my-profile'] })
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts [hunk 23]
```diff
@@ -0,0 +1,508 @@
...
+      )}`
+    );
+  });
+
+  test('Should validate Username Settings', async ({ page, browserName }) => {
+    test.skip(browserName === 'webkit', 'csrf_token cookie is being deleted');
+    await expect(
+      page
+        .locator('strong')
+        .filter({ hasText: translations.settings.labels.username })
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts [hunk 24]
```diff
@@ -0,0 +1,508 @@
...
+      ).toBeVisible();
+      await expect(
+        page.getByRole('link', {
+          name: `${translations.buttons['show-cert']} ${legacyCertifications[i]}`,
+          exact: true
+        })
+      ).toBeVisible();
+    }
+  });
+  test('Should validate Danger Section Settings', async ({
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts [hunk 25]
```diff
@@ -0,0 +1,508 @@
...
+    await expect(
+      page.getByText(
+        translations.settings.honesty.p7.replace(
+          settingsObject.supportEmailPlaceholder,
+          settingsObject.supportEmail
+        )
+      )
+    ).toBeVisible();
+  });
+  test('Should validate Certification Settings', async ({
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: done

File: e2e/settings.spec.ts [hunk 26]
```diff
@@ -0,0 +1,508 @@
...
+    );
+    await expect(addPortfolioButton).toBeVisible();
+    await addPortfolioButton.click();
+    const removeButton = page.getByRole('button', {
+      name: translations.buttons['remove-portfolio']
+    });
+    await expect(removeButton).toBeVisible();
+    await removeButton.click();
+  });
+  test('Should validate Accademy Honesty Settings', async ({
```
[the remark this answers is not in the document]
Comment (Riya267) [reply e2e/settings.spec.ts]: done

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍 Thanks @Riya267!

**Pull request merged**